EFCoreAssignment/Program.cs

```diff
@@ -20,30 +21,153 @@
 LoadingRelatedData_ExplicitLoading(dbContext);
 LoadingRelatedData_EagerLoading(dbContext);
 
+InsertProduct(dbContext);
+InsertProductWithNewShop(dbContext);
+UpdateProduct(dbContext);
+DeleteProduct(dbContext);
+DeleteProductByKey(dbContext);
+
+static void InsertProduct(AppDbContext dbContext)
+{
+    var product = new Product
+    {
+        Name = "McFlurry Oreo",
+        ShopId = 3
+    };
+
+    dbContext.Products.Add(product);
+    dbContext.SaveChanges();
+}
+
+static void InsertProductWithNewShop(AppDbContext dbContext)
+{
+    var shop = new Shop
+    {
+        Name = "Coco"
+    };
+
+    var product = new Product
+    {
+        Name = "Ice Water",
+        Shop = shop
+    };
+
+    dbContext.Shops.Add(shop);
+    dbContext.Products.Add(product);
+    dbContext.SaveChanges();
+}
+
+static void UpdateProduct(AppDbContext dbContext)
+{
+    var productId = 3;
+
+    var product = dbContext.Products.SingleOrDefault(p => p.Id == productId);
+    if (product != null)
+    {
+        product.Name = "Dog Food";
+        product.ShopId = 3;
+    }
+
+    dbContext.SaveChanges();
+}
+
+static void DeleteProduct(AppDbContext dbContext)
+{
+    var product = new Product
+    {
+        Id = 8,
+        Name = "Lechon",
+        ShopId = 2
+    };
+
+    dbContext.Products.Remove(product);
+    dbContext.SaveChanges();
+}
+
+static void DeleteProductByKey(AppDbContext dbContext)
+{
+    var productId = 14;
+
+    var product = dbContext.Products.SingleOrDefault(x => x.Id == productId);
```

Comment:
No need to query a **product** entity here. Just create a new instance of the entity and add/pass an Id.

Reply:
Updated code. Thanks!